### PR TITLE
feat(SFINT-5673): UA event logTriggerNotify added to insight client

### PR DESCRIPTION
**Pull request by SimonMilord**
### Proposed changes:

The following custom event has been added to Insight Client for notify triggers in insight usecase:

- logTriggerNotify

[SFINT-5673]: https://coveord.atlassian.net/browse/SFINT-5673?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/insight/insightClient.ts [hunk 1]
```diff
@@ -687,6 +688,10 @@ export class CoveoInsightClient {
         );
     }
 
+    public logTriggerNotify(metadata: TriggerNotifyMetadata) {
```
Comment (mmitiche) [inline src/insight/insightClient.ts:691]: We usually include the case metadata in the custom events logged for insight panel, here is an example: https://github.com/coveo/coveo.analytics.js/blob/984b149ae27f548417dd5a9d7153c7f1e188d703/src/insight/insightClient.ts#L595-L605

Comment (SimonMilord) [reply src/insight/insightClient.ts:691]: oh right, will add !

**Review by mmitiche — APPROVED** [review]
LGTM!

**Review by erocheleau — APPROVED** [review]
Is there an EP version of this?

**Comment by jstpierref** [comment]
> Is there an EP version of this?

@erocheleau No!

**Review by lbourdages — COMMENTED** [review]
Overall it looks okay but I want to challenge the addition of new stuff to this library which we're trying to get rid of. Shouldn't we at least make that available in EP as Étienne suggested?

**Comment by erocheleau** [comment]
> Overall it looks okay but I want to challenge the addition of new stuff to this library which we're trying to get rid of. Shouldn't we at least make that available in EP as Étienne suggested?

This already exists in UA, we're just adding it to our Insight version of the UA library because it was missing.

For EP we're still trying to figure out if we could perhaps rely on a server-side event instead of a client-side that's why we're not adding it to anything EP for now.

And we won't get rid of this library any time soon anyway :)

**Review by lbourdages — APPROVED** [review]

**Pull request merged**